### PR TITLE
notifications (and windows msi installer)

### LINKED ISSUE #203: Deskto notifications?
Hi, thanks for the great IRC client.

I was wondering if you were interested in adding support for desktop notifications? On Linux, there's the `libnotify` system, and a simple binary named `notify-send`. This would help with getting a message while working with another application, a common user behavior.

Thanks! :smile:

### PR DESCRIPTION
**Pull request by casperstorm**
This PR adds notification support, along a few events: `Connected`, `Reconnected` and `Disconnected`.
We should add more, but for now the base should be ready.

Reason this PR also includes a new installer for Windows is that in order to get a icon on the notification on Windows we need a `app_id`, and the easiest way (that i know of) is to do it through a `.wxs` file. For that reason we might as well produce a `.msi` installer as our "primary" windows installation. Going forward we could then use a default `.exe` asset for our "portable" installer (#141) - this would then be similar to how Alacritty does releases and assets. 

In any case, with the app installed through the `.msi` installer notifications has a small icon which looks great.

### Sounds
The configuration can take a string to a sound. Each OS has a different set of default sounds, depending on OS version as well. However in order for the user to know what to pick, we need to make a entry to our wiki with the different defaults and how to find them (see below).

This entire branch is currently untested on Linux - @tarkah could you test this?

#### macOS (`/System/Library/Sounds/`)
```
Basso
Blow
Bottle
Frog
Funk
Glass
Hero
Morse
Ping
Pop
Purr
Sosumi
Submarine
Tink
```

#### Windows (https://github.com/allenbenz/winrt-notification/blob/main/src/lib.rs#L70-L75)
```
Default
IM
Mail
Reminder
SMS
```

#### Unix (https://github.com/hoodie/notify-rust/blob/main/src/hints.rs#L68)
```
message-new-instant
[...]
```

Fixes #203

### CODE REVIEW

**Comment by casperstorm** [comment]
@tarkah what do you think we should add before merge?
- DM notification?
- Highlight notification?

File: data/src/client.rs [hunk 1]
```diff
@@ -57,6 +62,7 @@ pub enum Event {
     Single(message::Encoded, Nick),
     WithTarget(message::Encoded, Nick, message::Target),
     Brodcast(Brodcast),
+    Notification(Notification),
```
Comment (tarkah) [inline data/src/client.rs:65]: We need to add `message::Encoded` here so we still record the message in history once returned from here

File: data/src/client.rs [hunk 2]
```diff
@@ -683,6 +717,15 @@ impl Client {
                 Retry,
             }
 
+            match self.highlight_blackout {
+                HighlightBlackout::Blackout(instant) => {
+                    if now.duration_since(instant) >= HIGHLIGHT_BLACKOUT {
```
Comment (tarkah) [inline data/src/client.rs:722]: This can go above / outside the `for` loop

File: data/src/client.rs [hunk 3]
```diff
@@ -364,10 +374,19 @@ impl Client {
...
-                    if user.nickname() == self.nickname() && context.is_some() {
+                    // Highlight notification
+                    if message::reference_user(user.nickname(), self.nickname(), text)
+                        && self.highlight_blackout.allow_highlights()
+                    {
+                        events.push(Event::Notification(
+                            message.clone(),
+                            self.nickname().to_owned(),
+                            Notification::Highlight(user, channel.clone()),
+                        ));
```
Comment (tarkah) [inline data/src/client.rs:387]: Now that we include encoded message here, you can directly return here and remove the let events = vec![]; above

**Review by tarkah — APPROVED** [review]

**Pull request merged**